### PR TITLE
Auth system

### PR DESCRIPTION
**Pull request by salman-codewar**
## Purpose changes

Covers issue: #487 #480 

**Systems - Common**

Add Auth Provider in common, which will:
- [x]  Provide auth client instance
- [x] Provide authenticateUser method

**Systems - Auth/ApiGateway**

- [x] Auth system API GW added
- [x] Added Auth app redirect on swagger UI to et cookies in browser

**Systems - Auth Middleware integration**

- [x] Auth Middleware integration in Data Plan system
- [x] Auth middleware integration in Subscriber system
- [x] Auth middleware integration in Registry
- [x] Minor refactoring
- [x] Deployment

### CODE REVIEW

File: systems/data-plan/api-gateway/pkg/rest/router.go [hunk 1]
```diff
@@ -19,6 +20,8 @@ import (
 	"github.com/wI2L/fizz/openapi"
 )
 
+var REDIRECT_URI = "https://data-plan.dev.ukama.com/swagger/#/"
```
Comment (vthakur7f) [inline systems/data-plan/api-gateway/pkg/rest/router.go:23]: This doesn't look good. what happnes when somebody else deploy this for their org ?

**Review by vthakur7f — COMMENTED** [review]
Only thing I can point here is hardcoding of REDIRECT_URI add org name won't wokr for us. Lets get rid of these. After that we will be good to go.

File: systems/data-plan/api-gateway/pkg/rest/router.go (hunk 1 above)
Comment (salman-codewar) [reply systems/data-plan/api-gateway/pkg/rest/router.go]: Thanks for the feedback @vthakur7f , Addressed

**Review by vthakur7f — APPROVED** [review]
Nice Work..!!!

**Pull request merged**